udeploy-deployment-collector/src/main/java/com/capitalone/dashboard/collector/DefaultUDeployClient.java

```diff
@@ -144,55 +148,91 @@ public DefaultUDeployClient(UDeploySettings uDeploySettings,
  */
 
         // Failed to deploy list:
+        Set<String> failedComponents = getFailedComponents(nonCompliantResourceJSON);
+        Map<String, List<String>> versionFileMap = new HashMap<>();
+        for (Object item : allResourceJSON) {
```

Comment:
maybe if we load it into a model of some kind or a map it would be clearer whats going on rather than hand parsing it out.